source/Public/Install-SqlDscServer.ps1

```diff
@@ -0,0 +1,1398 @@
+            'SDK', # Part of parent feature Tools
+            'LocalDB', # Part of parent feature Tools
+            'ARC'
+        )]
+        [System.String[]]
+        $Features,
+
+        [Parameter(ParameterSetName = 'InstallRole', Mandatory = $true)]
+        [ValidateSet(
+            'ALLFeatures_WithDefaults',
+            'SPI_AS_NewFarm',
+            'SPI_AS_ExistingFarm'
+        )]
+        [System.String]
+        $Role,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'PrepareImage')]
+        [Parameter(ParameterSetName = 'InstallFailoverCluster')]
+        [Parameter(ParameterSetName = 'PrepareFailoverCluster')]
+        [System.String]
+        $InstallSharedDir,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'InstallFailoverCluster')]
+        [Parameter(ParameterSetName = 'PrepareFailoverCluster')]
+        [System.String]
+        $InstallSharedWowDir,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'PrepareImage')]
+        [Parameter(ParameterSetName = 'Upgrade')]
+        [Parameter(ParameterSetName = 'InstallFailoverCluster')]
+        [Parameter(ParameterSetName = 'PrepareFailoverCluster')]
+        [System.String]
+        $InstanceDir,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'PrepareImage', Mandatory = $true)]
+        [Parameter(ParameterSetName = 'CompleteImage')]
+        [Parameter(ParameterSetName = 'Upgrade')]
+        [Parameter(ParameterSetName = 'InstallFailoverCluster')]
+        [Parameter(ParameterSetName = 'PrepareFailoverCluster')]
+        [System.String]
+        $InstanceId,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'PrepareImage')]
+        [Parameter(ParameterSetName = 'CompleteImage')]
+        [Parameter(ParameterSetName = 'Repair')]
+        [Parameter(ParameterSetName = 'InstallFailoverCluster')]
+        [Parameter(ParameterSetName = 'PrepareFailoverCluster')]
+        [Parameter(ParameterSetName = 'AddNode')]
+        [System.String]
+        $PBEngSvcAccount,
+
+        [Parameter(ParameterSetName = 'Install')]
+        [Parameter(ParameterSetName = 'InstallRole')]
+        [Parameter(ParameterSetName = 'PrepareImage')]
+        [Parameter(ParameterSetName = 'CompleteImage')]
+        [Parameter(ParameterSetName = 'Repair')]
+        [Parameter(ParameterSetName = 'InstallFailoverCluster')]
+        [Parameter(ParameterSetName = 'PrepareFailoverCluster')]
+        [Parameter(ParameterSetName = 'AddNode')]
+        [System.Management.Automation.PSCredential]
+        $PBEngSvcPassword,
```

Comment:
We have to pass in a full PSCredential to just provide the password? Why not use `SecureString` or combine this as a PSCredential for `PBEngSvcAccount`? You can extract the username and password from the same object.

Reply:
This pattern could be used for each of the service accounts or username/password combos that have to be provided.

Reply:
We should use `SecureString`, I will update att password parameters.

I rather not combine the username and password into a `PSCredential` since then we then get problem with those that can take (g)MSA (then we must pass an empty/dummy password that need logic to be ignored).

Reply:
Changed to SecureString on all password parameters.